src/c-writer.cc

```diff
@@ -2472,13 +2472,23 @@ void CWriter::WriteImportProperties(CWriterPhase kind) {
 
   for (const Import* import : unique_imports_) {
     if (import->kind() == ExternalKind::Memory) {
-      const Limits* memory_limits =
-          &(cast<MemoryImport>(import)->memory.page_limits);
-      write_import_prop(import, "min", memory_limits->initial);
-      write_import_prop(import, "max",
-                        memory_limits->has_max
-                            ? memory_limits->max
-                            : std::numeric_limits<uint32_t>::max());
+      const Limits* limits = &(cast<MemoryImport>(import)->memory.page_limits);
+      // We use u64 so we can handle both 32-bit and 64-bit memories
+      const uint64_t default_max = limits->is_64
+                                       ? (static_cast<uint64_t>(2) << 48)
+                                       : (static_cast<uint64_t>(2) << 16);
```

Comment:
Maybe `static_cast<uint64_t>(1)` (or just `1UL`) instead of 2? E.g.  1 << 16 = 65,536.

Reply:
yikes, good catch. Unless there is a strong preference, I'll leave it as uint64_t though, as I think this keeps the intent clearer than UL.